src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,429 @@
+    EXPECT_NEAR(trajectory[num_segments - 1].position.x,
+                shared_polynomial2d_getValueOrder3(path, t_end).x, 0.01);
+    EXPECT_NEAR(trajectory[num_segments - 1].position.y,
+                shared_polynomial2d_getValueOrder3(path, t_end).y, 0.01);
+
+    EXPECT_NEAR(trajectory[0].position.x,
+                shared_polynomial2d_getValueOrder3(path, t_start).x, 0.01);
+    EXPECT_NEAR(trajectory[0].position.x,
+                shared_polynomial2d_getValueOrder3(path, t_start).y, 0.01);
+}
+
+TEST_F(TrajectoryPlannerTest, dynamics_dont_exceed_maximums_straight_line)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    const double t_start            = 0;
+    const double t_end              = 3;
+    const unsigned int num_segments = 500;
+    const double max_acc            = 3;
+    const double max_speed          = 9;
+    const double initial_speed      = 2;
+    const double final_speed        = 5;
+
+    TrajectoryElement_t* trajectory = generate_constant_arc_length_segmentation(
+        path, t_start, t_end, num_segments, max_acc, max_speed, initial_speed,
+        final_speed);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             __TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS__);
+
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(path, t_start, t_end,
+                                                        arc_length_param);
+
+    double velocities[num_segments];
+    double accelerations[num_segments];
+
+    std::vector<double> velocity =
+        getVelocityFromTrajectory(trajectory, num_segments, initial_speed);
+    std::vector<double> acceleration =
+        getAccelerationFromVelocity(velocity, trajectory, num_segments);
+
+    for (double vel : velocity)
+    {
+        EXPECT_TRUE(vel <= max_speed * 1.1 && vel >= initial_speed);
+    }
+
+    EXPECT_NEAR(velocity.back(), final_speed, 0.1);
+    EXPECT_DOUBLE_EQ(velocity.front(), initial_speed);
+
+    for (double acc : acceleration)
+    {
+        EXPECT_TRUE(fabs(acc) <= max_acc * 1.2);
+    }
+
+    EXPECT_NEAR(trajectory[num_segments - 1].position.x,
+                shared_polynomial2d_getValueOrder3(path, t_end).x, 0.01);
+    EXPECT_NEAR(trajectory[num_segments - 1].position.y,
+                shared_polynomial2d_getValueOrder3(path, t_end).y, 0.01);
+
+    EXPECT_NEAR(trajectory[0].position.x,
+                shared_polynomial2d_getValueOrder3(path, t_start).x, 0.01);
+    EXPECT_NEAR(trajectory[0].position.y,
+                shared_polynomial2d_getValueOrder3(path, t_start).y, 0.01);
+}
+
+TEST_F(TrajectoryPlannerTest,
+       path_length_and_start_point_end_point_test_reverse_direction)
```

Comment:
Could we make this test name a bit more descriptive? What are we checking for here?

Reply:
Changed to be more descriptive like above